pkg/systemd/terminal.jsx

```diff
@@ -26,17 +31,21 @@ const _ = cockpit.gettext;
      * Spawns the user's shell in the user's home directory.
      */
     class UserTerminal extends React.Component {
-        createChannel(user) {
-            return cockpit.channel({
+        createChannel(user, dir) {
+            const ch = cockpit.channel({
                 payload: "stream",
                 spawn: [user.shell || "/bin/bash"],
                 environ: [
                     "TERM=xterm-256color",
                 ],
-                directory: user.home || "/",
+                directory: dir || user.home || "/",
                 pty: true,
                 binary: true,
+                pid: null,
             });
+            ch.addEventListener("ready", (_, msg) => this.setState({ pid: msg.pid }), { once: true });
+            ch.addEventListener("close", () => this.setState({ pid: null }), { once: true });
```

Comment:
`once` doesn't actually work, please drop. Marius also said that the handlers clean up themselves once the object (channel in this case) gets destroyed, so should be fine.

Reply:
You resolved this thread, but it's still in the latest push. Reopening.